app/models/rule.rb

```diff
@@ -30,8 +30,8 @@ class Rule < ApplicationRecord
 
   has_many :profile_rules, dependent: :delete_all
   has_many :profiles, through: :profile_rules, source: :profile
-  has_many :rule_group_rules, dependent: :delete_all
-  has_many :rule_groups, through: :rule_group_rules
+  has_one :rule_group_rule, dependent: :delete
+  has_one :rule_group, through: :rule_group_rule
```

Comment:
Yes, this is the evil one that has to be remodeled :see_no_evil: `has_one through:` :laughing:

Reply:
Definitely 🙈 I already created the ticket 🙂